plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -449,7 +452,7 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
 
          auto after_sig_recovery =
                [self = this, trx, persist_until_expired, next{std::move(next)}]() mutable {
-                  app().post(priority::low, [self, trx{std::move(trx)}, persist_until_expired, next{std::move(next)}]() {
+                  self->_self->post(priority::low, [self, trx{std::move(trx)}, persist_until_expired, next{std::move(next)}]() {
                      self->process_incoming_transaction_async( trx, persist_until_expired, next );
```

Comment:
`chain::controller` is only safe to be accessed from the application thread. All calls into `controller` must be on the application thread.

`process_incoming_transaction_async` calls `chain.push_transaction` as well as other controller methods.